File: backend/src/domain/news/news.py
"""News domain — Entity + Value Object.

DDD roles:
- News: Entity (sportmonks-sourced).
- NewsType: Value Object (enum).
"""

from dataclasses import dataclass
from datetime import datetime
from enum import StrEnum


class NewsType(StrEnum):
    PREMATCH = "prematch"
    POSTMATCH = "postmatch"


@dataclass(frozen=True, slots=True)
class News:
    id: int
    fixture_id: int | None
    league_id: int | None
    title: str
    type: NewsType
    published_at: datetime | None = None

File: packages/core/src/domain/news/news.ts
// News domain — Sportmonks-sourced article tied (optionally) to a fixture.

export type NewsType = "prematch" | "postmatch";

export interface News {
  id: number;
  fixture_id?: number;
  league_id?: number;
  title: string;
  type: NewsType;
  published_at?: string;
  // Optional enrichment computed in the repo from cached fixtures/teams:
  // pretty label like "🇫🇷 France vs 🇦🇷 Argentina" so the UI doesn't need to
  // know about the relationships.
  fixture_label?: string;
}
